### DISCUSSION TITLE
Multiple AudioSourceStreams

### DISCUSSION THREAD

**Discussion by SaKiEQ** [pschatzmann/arduino-audio-tools#1838] [Q&A] [answered]
Hi all, 

How does one work with multiple AudioStreams, not necessarily in in parallel, but rather one after another and back ? 

Let's assume I want to have a welcome message played from a TTS source (MP3 Stream) and then switch to an URL Stream like an ICY source. Then if there are alerts back to a TTS source Stream. 

First off, do the different Stream Sources share the URLStream object (even if different format) ? 
Second how would one pause, stop, resume an AudioSourceURL stream ? 

```
URLStream urlStream(wifi, password); 

AudioSourceIcyUrl sourceStream(urlStream, urls, "audio/mp3");
AudioSourceURL    sourceTTS(urlStream, url_1, "audio/mp3");
```

Thanks in advance.

**Comment by pschatzmann** [comment, marked as answer]
I suggest you read the Introduction Wiki, then you should be able to answer this yourself!

**Reply by SaKiEQ** [reply]
Well I did read pretty much every page several times. But you logic is not necessarily compatible with everybody's else's logic. 
While I can easily follow all the basic principles outlined in the Wiki - I do am apparently incapably to follow the underlaying design philosophy or strategy. 

I think it's time to stop this project. 

Thanks.

**Comment by pschatzmann** [comment]
You can have multiple sources and sinks: You decide in the loop from which source you copy the data to what destination.

If the audio info is changing you can just call destination.setAudioInfo(source.audioInfo()) to synchronize the sample rate, channels and bits per sample or you can let the StreamCopy do this job by calling copier.setSynchAudioInfo(true) in the setup.

Finally, as a conclusion you stop, start resume the processing by just stopping or resuming to call copier.copy() or you can call copier.setActive(flag) with the flag true or false, to achieve the same thing.
